### PR TITLE
Add Guard, run tests all the time.

### PR DESCRIPTION
**Pull request by robbkidd**
Y'know, if you want to.

### CODE REVIEW

File: Guardfile [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    watch(rails.spec_helper)     { rspec.spec_dir }
+    watch(rails.routes)          { "#{rspec.spec_dir}/routing" }
+    watch(rails.app_controller)  { "#{rspec.spec_dir}/controllers" }
+
+    # Capybara features specs
+    watch(rails.view_dirs)     { |m| rspec.spec.call("features/#{m[1]}") }
+    watch(rails.layouts)       { |m| rspec.spec.call("features/#{m[1]}") }
+  end
+
+  guard :rubocop, all_on_start: true do
```
Comment (bjmllr) [inline Guardfile]:
```ruby
guard :rubocop, all_on_start: true, cli: '-aD' do
```

`-a` - autocorrect
`-D` - show cop names

Comment (bjmllr) [inline Guardfile]: I guess there's also a `--rails` flag to run rails-specific cops which we would put here. I've never used it.

File: Gemfile [hunk 2]
```diff
@@ -14,6 +14,9 @@ group :development, :test do
   gem 'better_errors'
   gem 'rspec-rails', '~> 3.0'
   gem 'rubocop'
+  gem 'guard'
```
Comment (bjmllr) [inline Gemfile]: Should this be in the `:test` group or just `:development`?

File: Guardfile (hunk 1 above)
Comment (robbkidd) [inline Guardfile]: I _think_ [these lines in `.rubocop.yml`](https://github.com/rubyforgood/habitat_humanity/blob/master/.rubocop.yml#L2-L3) do what `--rails` does.

File: Gemfile (hunk 2 above)
Comment (robbkidd) [inline Gemfile]: Prob just dev, yep.

**Pull request merged**